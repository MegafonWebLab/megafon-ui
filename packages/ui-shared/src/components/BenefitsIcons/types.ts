type TGridSizeValues = '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' | '10' | '11' | '12';

export interface IBenefit {
    /** Заголовок бенефита */
    title?: string;
    /** Основной текст в бенефите */
    text?: string;
    /** Иконка */
    icon: React.ReactNode;
}

export const IconPositionEnum = {
    LEFT_TOP: 'left-top',
    CENTER_TOP: 'center-top',
    LEFT_SIDE: 'left-side',
} as const;
export type IconPosition = typeof IconPositionEnum[keyof typeof IconPositionEnum];

export type GridConfig = {
    wide?: TGridSizeValues;
    desktop?: TGridSizeValues;
    tablet?: TGridSizeValues;
    leftOffsetWide?: TGridSizeValues;
    leftOffsetDesktop?: TGridSizeValues;
    leftOffsetTablet?: TGridSizeValues;
    rightOffsetDesktop?: TGridSizeValues;
    rightOffsetTablet?: TGridSizeValues;
};
