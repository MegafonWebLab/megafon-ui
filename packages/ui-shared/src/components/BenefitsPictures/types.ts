export interface IBenefit {
    /** Заголовок бенефита */
    title: string;
    /** Основной текст в бенефите */
    text: string;
    /** Изображение */
    img: string;
}

type TGridSizeValues = '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' | '10' | '11' | '12';

export type GridConfig = {
    wide?: TGridSizeValues;
    desktop?: TGridSizeValues;
    tablet?: TGridSizeValues;
    leftOffsetWide?: TGridSizeValues;
    leftOffsetDesktop?: TGridSizeValues;
    leftOffsetTablet?: TGridSizeValues;
    rightOffsetDesktop?: TGridSizeValues;
    rightOffsetTablet?: TGridSizeValues;
    rightOffsetWide?: TGridSizeValues;
};

export type GridGutterSize = 'large' | 'medium';
