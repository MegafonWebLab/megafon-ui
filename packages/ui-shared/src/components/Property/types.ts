export type Desc = {
    value: string[];
    isCollapsible?: boolean;
};

export type Item = {
    title?: string[];
    description?: Desc[];
    value?: string;
};
