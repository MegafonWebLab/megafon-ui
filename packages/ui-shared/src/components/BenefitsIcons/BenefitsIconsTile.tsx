import * as React from 'react';
import PropTypes from 'prop-types';
import convert from 'htmr';
import './style/BenefitsIconsTile.less';
import { Header, Paragraph, cnCreate } from '@megafon/ui-core';
import { IBenefit, IconPositionEnum } from './types';

export interface IBenefitsIconsTile extends IBenefit {
    iconPosition?: IconPositionEnum;
}

const cn = cnCreate('mfui-beta-benefits-icons-tile');
const BenefitsIconsTile: React.FC<IBenefitsIconsTile> = ({
    title,
    text,
    icon,
    iconPosition = IconPositionEnum.LEFT_TOP,
}) => (
    <div className={cn('', { 'icon-position': iconPosition })}>
        <div className={cn('svg-icon')}>{icon}</div>
        <div className={cn('content-wrapper')}>
            {title && (
                <Header className={cn('title')} as="h5">
                    {convert(title)}
                </Header>
            )}
            {text && (
                <Paragraph className={cn('text')} hasMargin={false}>
                    {convert(text)}
                </Paragraph>
            )}
        </div>
    </div>
);

BenefitsIconsTile.propTypes = {
    title: PropTypes.string,
    text: PropTypes.string,
    iconPosition: PropTypes.oneOf(Object.values(IconPositionEnum)),
    icon: PropTypes.node,
};

export default BenefitsIconsTile;
