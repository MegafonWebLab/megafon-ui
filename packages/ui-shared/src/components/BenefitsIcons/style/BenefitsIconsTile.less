@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-beta-benefits-icons-tile');

@{b} {
    margin-bottom: 48px;
    padding: 0 16px;

    @{b}__svg-icon {
        margin-bottom: 10px;
    }

    &__title + &__text {
        margin-top: 4px;
    }

    &__svg-icon {
        width: 44px;
        height: 44px;
        overflow: hidden;
    }

    &_icon-position_left-side {
        display: flex;

        @{b}__content-wrapper {
            padding-top: 10px;
        }

        @{b}__svg-icon {
            flex-shrink: 0;
            margin-right: 14px;
            margin-bottom: 0;
        }
    }

    &_icon-position_center-top {
        text-align: center;

        @{b}__svg-icon {
            margin-right: auto;
            margin-left: auto;
        }
    }
}
