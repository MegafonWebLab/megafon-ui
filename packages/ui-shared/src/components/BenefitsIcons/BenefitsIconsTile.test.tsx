import * as React from 'react';
import { shallow } from 'enzyme';
import { IconPositionEnum } from './types';
import BenefitsIconsTile, { IBenefitsIconsTile } from './BenefitsIconsTile';
import Checked from 'icons/System/24/Checked_24.svg';

const props: IBenefitsIconsTile = {
    title: 'title',
    text: 'text',
    iconPosition: IconPositionEnum.LEFT_SIDE,
    icon: <Checked />,
};

describe('<BenefitsIconsTile />', () => {
    it('renders BenefitsIconsTile', () => {
        const wrapper = shallow(<BenefitsIconsTile {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders BenefitsIconsTile without title', () => {
        const wrapper = shallow(<BenefitsIconsTile {...props} title={undefined} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders BenefitsIconsTile without text', () => {
        const wrapper = shallow(<BenefitsIconsTile {...props} text={undefined} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders BenefitsIconsTile with html in text field', () => {
        const newProps = {
            ...props,
            text: '<ul as="ol" color="red"><li><p>пункт 1</p></li><li>пункт 2</li><li><a>пункт</a> <b>3</b></li></ul>',
        };

        const wrapper = shallow(<BenefitsIconsTile {...newProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});
