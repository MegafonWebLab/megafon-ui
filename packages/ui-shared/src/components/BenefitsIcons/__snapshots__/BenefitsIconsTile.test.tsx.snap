// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BenefitsIconsTile /> renders BenefitsIconsTile 1`] = `
<div
  className="mfui-beta-benefits-icons-tile mfui-beta-benefits-icons-tile_icon-position_left-side"
>
  <div
    className="mfui-beta-benefits-icons-tile__svg-icon"
  >
    <test-file-stub />
  </div>
  <div
    className="mfui-beta-benefits-icons-tile__content-wrapper"
  >
    <Header
      as="h5"
      className="mfui-beta-benefits-icons-tile__title"
      color="black"
      hAlign="inherit"
    >
      title
    </Header>
    <Paragraph
      className="mfui-beta-benefits-icons-tile__text"
      hasMargin={false}
    >
      text
    </Paragraph>
  </div>
</div>
`;

exports[`<BenefitsIconsTile /> renders BenefitsIconsTile without text 1`] = `
<div
  className="mfui-beta-benefits-icons-tile mfui-beta-benefits-icons-tile_icon-position_left-side"
>
  <div
    className="mfui-beta-benefits-icons-tile__svg-icon"
  >
    <test-file-stub />
  </div>
  <div
    className="mfui-beta-benefits-icons-tile__content-wrapper"
  >
    <Header
      as="h5"
      className="mfui-beta-benefits-icons-tile__title"
      color="black"
      hAlign="inherit"
    >
      title
    </Header>
  </div>
</div>
`;

exports[`<BenefitsIconsTile /> renders BenefitsIconsTile without title 1`] = `
<div
  className="mfui-beta-benefits-icons-tile mfui-beta-benefits-icons-tile_icon-position_left-side"
>
  <div
    className="mfui-beta-benefits-icons-tile__svg-icon"
  >
    <test-file-stub />
  </div>
  <div
    className="mfui-beta-benefits-icons-tile__content-wrapper"
  >
    <Paragraph
      className="mfui-beta-benefits-icons-tile__text"
      hasMargin={false}
    >
      text
    </Paragraph>
  </div>
</div>
`;
